RFC 005 - Template contributing proposed process

barbaricyawps opened the pull request:
# Note to RFC reviewers

This pull request description is not the actual RFC. To view the actual RFC, click **Files changed** in the top menu for this pull request. NOTE: I recommend reading the files in their rendered format rather than their raw format.

## Current RFC status

- [x] Under discussion (until 2021-09-08)
- [x] Final comment and voting (until 2021-09-29)
- [ ] On hold
- [ ] Accepted
- [ ] Rejected
- [ ] Implemented
- [ ] Deferred
- [ ] Withdrawn

emckean commented on rfc-005-template-contributing-process.md:
```diff
@@ -0,0 +1,154 @@
+
+## Open questions
+
+We haven't really found a good title for the template deliverable I'm currently calling "Before you start." Cameron prefers "theory." Suggestions are appreciated! Other ideas that were floated in the community:
```
+1 for 'before you start': very clear and helpful!

gayathri-krishnaswamy replied:
I like Before You Start as well...

camerons replied:
thought (non-blocking): My reservations with "before you start" is that while this theory type doc is likely to be read before starting on a doc, I anticipate it will often be read as part of a **future** doc iteration, like a phase 2 improvement.

Another word I've recently [seen in the wild](https://www.w3.org/TR/skos-primer/) is a "primer". A primer is a conceptual document which provides the background theory related to a topic. While a primer need not be read "before you start" is does imply you should read first (like a primer coat of paint, being the first coat of paint you use). How would you feel about "primer"?

For reference on what might go into the "before you start"/theory/primer, you can skim this early version:
https://github.com/thegooddocsproject/templates/blob/dev/base/base-theory.md

barbaricyawps replied:
I do like primer. That could possibly work well because it doesn't sound as heavy as "theory," doesn't pre-suppose when you will read the doc, and doesn't make you instantly forecast what the content is going to be. Other reactions?

mgan59 replied:
I like `primer` as well.

aidanatwork replied:
I prefer "About this Template" or just "About". Filename could be ***-about.md. I find Before you Start problematic because in many cases content like references supporting the decisions made to construct the template are simply not of interest to end users, and so it is incorrect to say that they need to review them before they use the template. Some users will want this, and perhaps even more common will be new templateers interested in deciding for themselves if they agree with the approach taken when creating the template originally.  Essential facts that the user should know before filling out the template should be in ***-guide, not in some other file. Primer also seems incorrect here --- it refers to an introductory educational book about a subject. The scope of what's being described is too small for primer to make sense. My understanding also is that knowledge essential to read before using the template should go in ****-guide. There seems to be quite a bit of overlap here about what the ***-guide is vs. ****-theory/before-you-start. I would rate this a blocking concern.

viraji replied:
Out of the already proposed names, I vote for “Template background”. But I would also like to suggest “Backdrop” or “Groundwork” implying just that. Because this is likely to contain all the information that went into setting the stage for the actual template.

johns10 replied:
I don't have a strong opinion here, but "Getting Started" is a common term used for this or something like it, and people will know what it means.

barbaricyawps replied:
@aidanatwork , I see your point, but remember that the description for this content is "This optional guide is designed to help template users understand some of the key research, brainstorming, or pre-writing steps that they might need to do before adding this type of document to a guide. If this documentation type has certain key decisions that need to be made with stakeholders or if it has a dependency on some other type of document or process, that should be mentioned in this guide."

For that reason, I don't feel like "about" accurately describes what this document does.

For some extra context, the purpose of this additional optional document was discussed heavily in the Community Docs working group on August 8 and this was the decision that the 5 templateers in attendance gave. (See [Community Docs notes](https://docs.google.com/document/d/1iRxB0T7FoVjIeG_LawbCzcLfXCILlpFdzn-4BPDGR9Y/edit?usp=sharing) for more context.) Here's the key takeaways:

- Re: the base template "theory" document: The general recommendation was to include most of this material in a simpler, more lightweight form in the template guide document. The template guide document could open with a section about “Why do I need this type of document”, then it could give a guide for filling in all the sections of the document, then it could close with a list of links or references for deeper learning and research (including any Good Docs Project blog entries or white papers).
- We also had a brief conversation about content that seems to be missing: what about a guide that helps you do “pre-writing” and brainstorming or work through some of the research you need to do to prepare to write the template? Perhaps we could create a simple guide that asks you questions and gets you to think through what needs to be considered before you begin writing. This content could possibly go into the template guide (but that guide is starting to have a lot of content, isn’t it?) or it could be a separate document.

That second comment forms the basis for why it was originally called "before you start." It was meant to include pre-writing activities or questions to help you do the necessary groundwork you needed to do before you even sat down to write.

For that reason, I am still somewhat partial to "Getting Started" or "Before You Start" but I am also open to "Primer."

barbaricyawps replied:
Thanks to everyone who commented on the naming and content of the "before you start" document. We discussed this at length at last week's general meeting and the group consensus there was to call this the "deep dive" document. It basically contains miscellaneous information or guidance that is too comprehensive for the "guide" document. Based on whatever the each unique template needs, this document could include research, deep theories, pre-writing information, and/or any other material that users need to know when implementing this topic. Template writers are encouraged to point/link readers to that document in their guide to encourage users to read it in more depth.

I'm adding this content to the template documents and, when that is done, I will resolve these comments.

aidanatwork commented on template-deliverables.md:
```diff
@@ -0,0 +1,145 @@
+    <td>Yes, required</td>
+  </tr>
+  <tr>
+    <td>Before you start guide</td>
```
there may be some conceptual overlap here between Before you start and Guide.  If something needs to be understood before using the template, I'd think that should go in Guide. Stuff like references to support the decisions made in creating the template do not necessarily need to be understood "before you start". I had thought of this file more like a "Reference" or "About this template" file.

barbaricyawps replied:
See my reply to your previous comment.

barbaricyawps replied:
Resolved in our US general weekly meeting: We discussed this at length at last week's general meeting and the group consensus there was to call this the "deep dive" document. It basically contains miscellaneous information or guidance that is too comprehensive for the "guide" document. Based on whatever the each unique template needs, this document could include research, deep theories, pre-writing information, and/or any other material that users need to know when implementing this topic. Template writers are encouraged to point/link readers to that document in their guide to encourage users to read it in more depth.

gayathri-krishnaswamy commented on rfc-005-template-contributing-process.md:
```diff
@@ -0,0 +1,154 @@
+
+3. **Research and draft** - They research best practices and look at examples of the type of document they are working on. When they are finished, they will create a rough draft.
+
+4. **Workshop the drafts with the community** - Revise and refine their template set with the community pull reviewers. At least three community reviews are required to move to the next phase.
```
Can we reword Workshop the drafts to Review the drafts?  My understanding is anybody in the community can review the draft or is it only community pull reviewers? I think we need more clarity on "three community reviews are required". I am unable to figure what that means. Is it 3 rounds of review? Wondering if we can provide information on how the review happens. It can be over the slack or the weekly community meetings

aidanatwork replied:
I agree that to make this more clear, we could just say "Revise drafts based on community review". At least three community reviews sounds to me like at least three people need to review, not three rounds of review, but I agree it is unclear. Three rounds of review seems like an excessively high bar. As for how the review happens, I would assume it is simply reading the drafts and leaving comments, not likely something in real-time.

johns10 replied:
My impression is that "Workshop" refers to actually using the template to create documents, not reviewing the actual template. My assumption is that the reviewing will happen in 1-3, and that "workshopping" is more than just reviewing and providing feedback.

camerons replied:
As I've seen our community working so far, our working groups have been collaborating and discussing a template in real time, which is closer to a "workshop" concept. I see "review" as something which is time-shifted and a separate concept. Hence my mild preference for retaining "workshop" over "review".
+1 for clarifying "reviews" and "reviewers".
Like Aiden, I've questioned whether 3 reviewers is too high a quality bar. (Is our community big enough to support this?) Maybe drop to 2. However I don't feel strongly about this, and think that 3 is fine.

barbaricyawps replied:
Yes, for clarification, it's not 3 rounds of reviews. It's that at least 3 different community reviewers will provide feedback on the draft. 3 community reviewers are necessary to advance to the next stage. I'll make sure this section is revised to clarify the nature of the requirement.

Can anyone suggest a simple, succinct verb to replace "workshop" that encompasses the meaning here?

barbaricyawps replied:
Thanks to everyone who commented on this part of the RFC. We had a lengthy discussion about it at our last general meeting. The consensus was that the term "workshop," although accurate, was not necessarily understood across all cultures. For that reason, we're going to change the name of this phase to "Get feedback on drafts from the community." There was also some confusion about who could review these drafts. We'll clarify that community reviewers includes anyone who is a member of the Good Docs Project, including members of template writing working groups. For now, we'll keep the requirement to have at least three community reviews are required to move to the next phase. We'll adjust this requirement if we discover it is too difficult.

gayathri-krishnaswamy commented on template-deliverables.md:
```diff
@@ -0,0 +1,145 @@
+- **The template title** - The type of document this template is for. For example, "Tutorial template" or "README template."" This title should be formatted with the heading 1 weight.
+- **Introductory comment** - Include an embedded writing tip at the start that tells users they should first read the accompanying template guide and before-you-start guide before they fill in the template. Put introductory comments in {curly brackets}.
+- **The template sections** - Each section of the temple should begin with a heading with the heading 2 weight. The recommended content and some embedded writing tips fall under the headings.
+- **Embedded writing tips** - You can provide some lightweight writing tips that provide some tips or hints about what kind of content the template user might choose to put in a section of the document. Put embedded writing tips in {curly brackets}.
```
Can we give some examples for these embedded writing tips?

aidanatwork replied:
The most common would be after specifying the name of a section, have a writing tip that explains what kind of info ought to go in that section. For example:
"
h2. Request Parameters
{Writing tip: This is where you should list the inputs that end users can provide in the request payload. The simplest way to do this is with a table.}
"

camerons replied:
gayathri-krishnaswamy: We do intend to provide a [base template](https://github.com/thegooddocsproject/templates/blob/base-restructure3/base/base-template.md) which includes such examples and as such, I'd suggest that it doesn't need to be included here.

barbaricyawps replied:
Thanks for providing the example, Aidan.

One of the open action items is to create a template style guide that will include guidance about how to format embedded writing tips.

camerons commented on rfc-005-template-contributing-process.md:
```diff
@@ -0,0 +1,154 @@
+
+Each template set will include:
+- **Raw template** - The basic template for the type of document you are creating. It provides a rough outline of the suggested content and a few embedded writing tips for how to fill in the different sections of the template.
+- **Template guide** - Provides a deeper explanation of how to fill in the raw template. It provides a lightweight introduction to the purpose of this documentation and explains how to fill in each section of the document. Additional content may be included as needed.
```
suggestion (non-blocking): This text is based on our earlier thinking re split between "guide" and "before you start".  From more recent discussions, I think we having been coming to the opinion that we should push a bit more content from guide => template, and from "before you start" => guide.
Current wording is good enough for the moment, and we may refine after testing this with some real doctypes.

barbaricyawps replied:
Thanks, Cameron. I'll note this in the feedback section of the RFC.

gayathri-krishnaswamy commented on template-quality-checklist.md:
```diff
@@ -0,0 +1,102 @@
+- [ ] Raw template is present.
+- [ ] Template guide is present.
+- [ ] Template before-you-start guide is present (optional).
+- [ ] Template example is present OR a new issue was created to track this task (include link).
```
Should we indicate that template example is optional?

barbaricyawps replied:
Yes, I'll add that to the draft.

barbaricyawps replied:
I've adopted this suggestion.

acpkendo commented on template-contributing-guide.md:
```diff
@@ -0,0 +1,341 @@
+    <td><ul>
+          <li>Join a template writing working group led by a template mentor.</li>
+          <li>Work with a template mentor and your working group to decide which template you’ll work on.</li>
+          <li>Assign yourself to the corresponding issue for that template.</li>
```
As it stands right now I'd expect templates to be represented by epics, either release-specific or otherwise. We may need to adjust this wording accordingly.

barbaricyawps replied:
Agreed. Do we need to resolve this before the RFC is adopted or can this be revised after the fact?

barbaricyawps replied:
I'm not going to take action on this comment because it's out of the scope of this RFC. Creating the template project management system will be a separate RFC. I did note this in the feedback section of the RFC, though.

acpkendo commented on template-contributing-guide.md:
```diff
@@ -0,0 +1,341 @@
+In this phase, you will decide which template project you will work on and adopt that template.
+
+Be aware that:
+- Each template project is represented by a corresponding issue in the [Good Docs Project templates](https://github.com/thegooddocsproject/templates/issues) repository.
```
Issue or epic, TBD.

barbaricyawps replied:
See my reply to your previous comment.

acpkendo commented on template-contributing-guide.md:
```diff
@@ -0,0 +1,341 @@
+
+3. Notify the templates coordinator or your template mentor, who will help you get reviews for your draft from the community reviewers group.
+
+The templates coordinator will help ensure you get at least three high quality reviews from the community reviewers. When you’ve received sufficient community input and incorporated suggestions into your draft, you will collaborate with a templates coordinator to determine that your draft is ready to move to the next phase.
```
I'm a little worried about the level of coordination that might be required here, especially when considering iterations on templates post-v1, where each minor change requested might be represented by an issue, each of which would need the required three reviews.

aidanatwork replied:
Maybe two reviewers needed for a new template, and one reviewer needed for an update to an existing template?

barbaricyawps replied:
Keep in mind that we already have a requirement to author templates as part of a template-focused working group. So if you could even just get three community reviewers from your group, that satisfies the requirement. Our two template-focused working group currently both have well above three members. And then you always have the larger community to draw upon, of which we have many more than three.

barbaricyawps replied:
In last week's general meeting, we decided that, for now, we'll keep the requirement to have at least three community reviews are required to move to the next phase. We'll adjust this requirement if we discover it is too difficult.

camerons replied:
Non blocking comment: My memory of this meeting was that we wouldn't make "three" a hard requirement, but rather a strong recommendation.
```suggestion
The templates coordinator will help you find reviewers. Typically three or more community reviewers will provide a detailed review of your material. When you’ve received sufficient community input and incorporated suggestions into your draft, you will collaborate with a templates coordinator to determine that your draft is ready to move to the next phase.
```

aidanatwork commented on rfc-005-template-contributing-process.md:
```diff
@@ -0,0 +1,154 @@
+- **Template example (optional)** - The Good Docs Project is in the process of designing a fake documentation project that can provide examples of our templates in action. Until this project is ready, template authors may optionally create an example of the template in action.
+
+
+## Links and prior art
```
Perhaps this is just a term I'm unfamiliar with, but I'm not sure what "prior art" means in this context. "Prior work"?

camerons replied:
Nice catch Aidan. I've been seeing a mind-shift in technical writing away from common colloquial terms like "prior art" toward more exact terms, such as "prior work". Drivers for this are:
1. Helping internationalize documentation, so it is understandable by non-native speakers, and so the documentation is easier to translate.
2. Move to inclusive language, disassociating black/white with good/bad and replacing blacklist/whitelist to allowlist/denylist.

barbaricyawps replied:
This term was added to the RFC template per RFC-000 at Morgan's suggestion. I believe the term "prior art" comes from US patent law and generally refers to any kind of designs or other intellectual property that contributed to the final proposal.

barbaricyawps replied:
I'm not going to take action on this comment because it's not relevant to the scope of the RFC-005 and is about the RFC process itself. My recommendation would be that if we have objections to the term "prior art" that one of you opens a pull request to change the RFC template.

aidanatwork commented on template-contributing-guide.md:
```diff
@@ -0,0 +1,341 @@
+
+## Before you start
+
+Before starting, ensure you are familiar with The Good Docs Project’s goals, key concepts, and supported personas. For more information, see [About the Good Docs Project](https://thegooddocsproject.dev/about/).
```
This link does not appear to have any information about personas. If we are arguing that someone needs to be familiar with our personas before reading this, we should link directly to info about the personas. I'm not convinced, however, that knowing about our personas is a requirement for reading through the contributing process. That's probably something that can be introduced to new members in the writing working groups. Or we can link to it as part of the contributing guide.

johns10 replied:
I don't think we need to say anything about personas at this point. I'm not missing any context by not knowing about personas.

barbaricyawps replied:
Fair enough. I'll remove "personas."

barbaricyawps replied:
I've adopted this suggestion and I'm resolving this comment.

aidanatwork commented on template-contributing-guide.md:
```diff
@@ -0,0 +1,341 @@
+In addition to these roles, these resources are available to help you as you contribute templates to our project:
+
+- **Working groups** - Our project is organized into several working groups that meet on a regular basis to work on the project’s key initiatives. To effectively contribute to a project, consider joining one of these working groups.
+- **Templates issue list** - All the templates that are actively being worked on or which could be worked on have a corresponding issue in the templates repository. Use the issue list or the kanban board to find a template to work on and track your template’s progress as it moves throughout the template writing phases.
```
It would be good to mention that you can find all the template writing issues by sorting on the "templates" label.

barbaricyawps replied:
We'll have to revise this document after the fact as our project management tools and processes change.

aidanatwork commented on template-contributing-guide.md:
```diff
@@ -0,0 +1,341 @@
+    <th>Required?</th>
+  </tr>
+  <tr>
+    <td>Raw template</td>
```
Raw template does not seem like the right term for this file in the template set. Why not just call it "Template" or "Template File"? Raw implies there is some sort of refinement that has not happened yet. But this file will still be included in the template set after it has undergone all community feedback and revisions. In what sense is it "raw" at that point?

barbaricyawps replied:
Sure, that was just the wording used in the base template, but I'm fine with just "template" or "template file."

barbaricyawps replied:
I've adopted this suggestion. Resolving the comment.

aidanatwork commented on template-contributing-guide.md:
```diff
@@ -0,0 +1,341 @@
+  </tr>
+  <tr>
+    <td>Template guide</td>
+    <td>This guide provides a deeper explanation of how to fill in the raw template. It provides a lightweight introduction to the purpose of this documentation and explains how to fill in each section of the document. Additional content may be included as needed.</td>
```
If anything should be called "Before you start", shouldn't it be the guide? I actually think guide is a good name, but note from the description here that this is the essential info the user needs before they fill in the template.

barbaricyawps replied:
See my reply to your previous comment.

barbaricyawps replied:
I've updated our document based on what we agreed to in last week's US general meeting. Resolving.

aidanatwork commented on template-contributing-guide.md:
```diff
@@ -0,0 +1,341 @@
+  </tr>
+  <tr>
+    <td>Before you start guide</td>
+    <td>This optional guide is designed to help template users understand some of the key research, brainstorming, or pre-writing steps that they might need to do before adding this type of document to a guide. If this documentation type has certain key decisions that need to be made with stakeholders or if it has a dependency on some other type of document or process, that should be mentioned in ...
```
If this info is essential to using the template, it should be in ***-guide. I don't see a coherent explanation here distinguishing what goes in guide vs. what goes in **-before-you-start. There seems like considerable overlap to me..

barbaricyawps replied:
See my reply to your previous comment.

barbaricyawps replied:
I've updated our document based on what we agreed to in last week's US general meeting. Resolving.

aidanatwork commented on template-contributing-guide.md:
```diff
@@ -0,0 +1,341 @@
+    </td>
+  </tr>
+  <tr>
+    <td>4 - Workshop the drafts with the community</td>
```
I'd change this to "Revise the drafts based on community review"

barbaricyawps replied:
See my reply to your previous comment.

barbaricyawps replied:
I'm adopting your suggestion, per our discussion in last week's general meeting. Resolved: We had a lengthy discussion about it at our last general meeting. The consensus was that the term "workshop," although accurate, was not necessarily understood across all cultures. For that reason, we're going to change the name of this phase to "Get feedback on drafts from the community." There was also some confusion about who could review these drafts. We'll clarify that community reviewers includes anyone who is a member of the Good Docs Project, including members of template writing working groups. For now, we'll keep the requirement to have at least three community reviews are required to move to the next phase. We'll adjust this requirement if we discover it is too difficult.

aidanatwork commented on template-contributing-guide.md:
```diff
@@ -0,0 +1,341 @@
+  <tr>
+    <td>4 - Workshop the drafts with the community</td>
+    <td><ul>
+          <li>Revise and refine the drafts of your deliverables with help from at least three community reviewers.</li>
```
Since this is a requirement, I'd make it more clear what is required. Three community members must review the draft and provide feedback. The templateer should revise their draft based on the feedback they receive (if any).

barbaricyawps replied:
Agreed.

barbaricyawps replied:
I've adopted this suggestion and have made revisions to this part of the guide. Resolving.

aidanatwork commented on template-contributing-guide.md:
```diff
@@ -0,0 +1,341 @@
+  <tr>
+    <td>6 - Improve the template with user feedback</td>
+    <td><ul>
+          <li>After completing the previous phase, your template will officially be part of the Good Docs Project and will be available to our users.</li>
```
We might want to add a requirement here to create an issue to handle feedback and do the next version of the template, after it is finalized. This can then be assigned directly to the creator, or left unassigned (surfacing clearly that there is a gap in coverage, to facilitate governance of the content). But perhaps the template coordinator or mentors could do that.

barbaricyawps replied:
Let's leave this as an open action item that needs to be resolved when we work out the finer details of the template project management process.

barbaricyawps replied:
I've added your comment to the feedback section of the RFC.

aidanatwork commented on template-contributing-guide.md:
```diff
@@ -0,0 +1,341 @@
+In this phase, you will decide which template project you will work on and adopt that template.
+
+Be aware that:
+- Each template project is represented by a corresponding issue in the [Good Docs Project templates](https://github.com/thegooddocsproject/templates/issues) repository.
```
Good place to mention the `templates` label.

barbaricyawps replied:
We'll have to revise this document after the fact as our project management tools and processes change.

aidanatwork commented on template-contributing-guide.md:
```diff
@@ -0,0 +1,341 @@
+
+1. Scroll through the list of template issues and see if one interests you and/or matches your skill set. Alternatively, if you have an idea for a template project that does not yet have an issue, and you have the support of a template mentor, you can create a new issue for your template project.
+
+2. Assign yourself to the issue. See Assigning issues and pull requests in the GitHub docs for more information.
```
Wouldn't hurt to add the link: https://docs.github.com/en/issues/tracking-your-work-with-issues/assigning-issues-and-pull-requests-to-other-github-users

barbaricyawps replied:
Good call. I'll add that.

barbaricyawps replied:
I've adopted this suggestion. Resolving.

aidanatwork commented on template-contributing-guide.md:
```diff
@@ -0,0 +1,341 @@
+- **Looking at lots of examples.** Start by searching for examples of that type of document they want to create a template for. The more examples you can look at, the better. While it’s better to review good examples of that type of document, there is actually a lot of value in reviewing bad examples too. Consider keeping a spreadsheet to track which examples you used, what elements each one had  ...
+- **Searching for guides, books, blog posts, conference presentations, or videos about best practices.** Search the Internet to find advice, tips, or expert research about how to create that type of document. Consider posting in a forum for resource ideas. For example, asking for helpful guides or insights on a community forum like the Write the Docs Slack workspace could be beneficial. Be mindfu ...
+- **Reaching out to experts.** When you find people you admire, who have researched your topic already, try reaching out to them. They often have a “how to contact me” webpage. Ask if they’d be okay with their material being used. (They might need to republish under a different copyright.) Invite them to participate in the template working group. They might even be prepared to lead it. If you fee ...
+- **Collaborating with others in a working group.** Consider joining one of the template-focused working groups to meet regularly with a workshop of writers and discuss research ideas and findings.
```
The phrasing here is "consider". But elsewhere it seemed more like a hard requirement that template authors join a template-writing working group. We should try to make more clear whether or not that is a hard requirement.

barbaricyawps replied:
It's a hard requirement and this reflects that our thinking about making it a requirement shifted over the course of drafting this guide. I'll update this to communicate that it's a hard requirement.

barbaricyawps replied:
I've adopted this suggestion.

aidanatwork commented on template-contributing-guide.md:
```diff
@@ -0,0 +1,341 @@
+
+In this phase, you’ll begin to share your draft with community reviewers and invite feedback.
+Optionally, you might also consider sharing it beyond our community with other technical writing communities such as Write the Docs or beyond.
+The workshop phase is arguably the most crucial and important phase in the template writing process, so your template project might spend the bulk of its time in this phase.
```
We might want to rephrase this to avoid the term "to workshop". Perhaps call this the "feedback phase" or the "review phase".

barbaricyawps replied:
I don't like "feedback" because it isn't a verb the way the other phases are. "Review" could work though.

barbaricyawps replied:
Yes, we've reached a decision about this in the US general meeting last week. Resolved: We had a lengthy discussion about it at our last general meeting. The consensus was that the term "workshop," although accurate, was not necessarily understood across all cultures. For that reason, we're going to change the name of this phase to "Get feedback on drafts from the community." There was also some confusion about who could review these drafts. We'll clarify that community reviewers includes anyone who is a member of the Good Docs Project, including members of template writing working groups. For now, we'll keep the requirement to have at least three community reviews are required to move to the next phase. We'll adjust this requirement if we discover it is too difficult.

aidanatwork commented on template-contributing-guide.md:
```diff
@@ -0,0 +1,341 @@
+
+## Improve the template with user feedback
+
+After you’ve submitted your pull request in the previous phase, your template will officially be part of the Good Docs Project. Yay!
```
Is this true? Or are they officially part of the Good Docs Project after the PR is approved, rather than merely submitted?

johns10 replied:
The way it's described is as if the pull request is a formality. Should we smash in the reviews here? 

I think the reason that it's organized this way is because it takes work to get the template from the googly doc to the pull request, so we don't want people to do the work, then have to go back to the drawing board, change it, and then update in the PR?

barbaricyawps replied:
Well, it's when it's merged, so I'll clarify that.

We want to give people the option to do a Google Doc (because it's easier to review) or markdown. It's up to the template author.

barbaricyawps replied:
I've adjusted some of the text here to add more clarification that this is a final quality check with experienced project leaders doing the final review. Resolving this comment thread.

aidanatwork commented on template-deliverables.md:
```diff
@@ -0,0 +1,145 @@
+## Table of contents
+
+- [Overview](#overview)
+- [Raw template](#raw-template)
```
I think we should call this Template or Template File, for reasons I explain in more detail in a previous comment.

camerons replied:
I have a mild preference for "raw template" which distinguishes it from when we refer to the entire template directory as "template".

barbaricyawps replied:
Sure.

barbaricyawps replied:
I've adopted Aidan's suggestion. Resolving.

aidanatwork commented on template-deliverables.md:
```diff
@@ -0,0 +1,145 @@
+- [Overview](#overview)
+- [Raw template](#raw-template)
+- [Template guide](#template-guide)
+- [Before you start guide (optional)](#before-you-start-guide-optional)
```
I have some issues with calling this "Before you start guide", as well as more generally about what the purpose of this file is. More detail in previous comments.

barbaricyawps replied:
See my reply to your previous comments.

barbaricyawps replied:
I've updated this section based on the consensus we reached at the US general meeting last week: We discussed this at length at last week's general meeting and the group consensus there was to call this the "deep dive" document. It basically contains miscellaneous information or guidance that is too comprehensive for the "guide" document. Based on whatever the each unique template needs, this document could include research, deep theories, pre-writing information, and/or any other material that users need to know when implementing this topic. Template writers are encouraged to point/link readers to that document in their guide to encourage users to read it in more depth.

aidanatwork commented on template-deliverables.md:
```diff
@@ -0,0 +1,145 @@
+  </tr>
+  <tr>
+    <td>Raw template</td>
+    <td>The raw template is the basic template for the type of document you are creating. It provides a rough outline of the suggested content and a few embedded writing tips for how to fill in the different sections of the template.</td>
```
Calling this the "raw template" or the "basic template" unintentionally implies there might be other templates in the template sets. There is just one, and this is it. It is the template itself. Perhaps this should read "The template file itself contains a suggested content structure, with fields that a user can replace to generate a document of this type." Something like that.

barbaricyawps replied:
Sure.

barbaricyawps replied:
I've adopted this suggestion into the draft. Resolving.

aidanatwork commented on template-deliverables.md:
```diff
@@ -0,0 +1,145 @@
+Each raw template should have this content:
+- **The template title** - The type of document this template is for. For example, "Tutorial template" or "README template."" This title should be formatted with the heading 1 weight.
+- **Introductory comment** - Include an embedded writing tip at the start that tells users they should first read the accompanying template guide and before-you-start guide before they fill in the template. Put introductory comments in {curly brackets}.
+- **The template sections** - Each section of the temple should begin with a heading with the heading 2 weight. The recommended content and some embedded writing tips fall under the headings.
```
We should not limit the template writer to header 2 weight, as they may have sub-sections within each section that merit a lower weight.

camerons replied:
Suggestion: I agree with Aidan, that it may be helpful to comment on subsections. Maybe something like:
"Occasional use of sub-sections / sub-headings may be used if required. However, note that deep use of heading structures is usually an indication that a content restructure should be considered."

barbaricyawps replied:
Maybe I meant to say 3. I'll give it some thought.

barbaricyawps replied:
It's a little buried, but I do say they can use heading 3 in one of the bullet points. I'll resolve this comment for now. We do need to create a style guide, though.

aidanatwork commented on template-deliverables.md:
```diff
@@ -0,0 +1,145 @@
+This guide provides a deeper explanation of how to fill in the raw template.
+It provides a lightweight introduction to the purpose of this documentation and explains how to fill in each section of the document.
+
+In essence, this guide supports the key decisions that the template user might need to make during the writing process.
```
Change to: "This guide explains the key decisions that the template user needs to make during the writing process."

barbaricyawps replied:
Sure.

barbaricyawps replied:
I've adopted this suggestion.

aidanatwork commented on template-deliverables.md:
```diff
@@ -0,0 +1,145 @@
+
+## Before you start guide (optional)
+
+Example filename: before-you-start-abc-template.md
```
I recommend we stick the convention of having the template name at the beginning of each file, and the words that distinguish this file from others in the template set at the end. 

So that would mean this should be something like: abc-template-before-you-start.md

barbaricyawps replied:
This is where "primer" might be better.

barbaricyawps replied:
I've adopted this suggestion.

aidanatwork commented on template-deliverables.md:
```diff
@@ -0,0 +1,145 @@
+
+Example filename: before-you-start-abc-template.md
+
+This optional guide is designed to help template users understand some of the key research, brainstorming, or pre-writing steps that they might need to do before adding this type of document to a guide.
```
This all seems to be covered by the guide file. Even more confusingly, this file is referred to as a "guide" in several places. If we keep both this file and the guide file with their current names, we should avoid referring to this file as a guide to avoid confusion.

camerons replied:
I suggest that we expand our target audience here to:
* Docset owner, building the information architecture for a project. This person may be a technical writer.
* Curious and engaged author, who wants to know more than "just follow the steps in the guide".
* Not your "just do write a doc" author.
Note: We've moved back and forth on deciding whether sections fit in the guide or theory/primer doc.

barbaricyawps replied:
See my reply to your previous comment.

barbaricyawps replied:
I've revised this section based on what we agreed in last week's general meeting. Resolved: We discussed this at length at last week's general meeting and the group consensus there was to call this the "deep dive" document. It basically contains miscellaneous information or guidance that is too comprehensive for the "guide" document. Based on whatever the each unique template needs, this document could include research, deep theories, pre-writing information, and/or any other material that users need to know when implementing this topic. Template writers are encouraged to point/link readers to that document in their guide to encourage users to read it in more depth.

aidanatwork commented on template-quality-checklist.md:
```diff
@@ -0,0 +1,102 @@
+
+### Procedural requirements
+
+- [ ] The template contributor participated in a template working group to create this template.
```
Might be good to have them list which group they were part of.

barbaricyawps replied:
Sure.

barbaricyawps replied:
I've adopted this suggestion.

aidanatwork commented on template-quality-checklist.md:
```diff
@@ -0,0 +1,102 @@
+### Procedural requirements
+
+- [ ] The template contributor participated in a template working group to create this template.
+- [ ] This template was workshopped with the community.
```
I'd avoid using the term workshopped and say "This template was reviewed by at least three community members." Though as noted in one of my other comments, I think two members is sufficient for a new template and one member sufficient for a revision to an existing template.

barbaricyawps replied:
Noted, but I prefer 3. See my previous comment.

barbaricyawps replied:
I've adopted this suggestion.

aidanatwork commented on template-quality-checklist.md:
```diff
@@ -0,0 +1,102 @@
+
+- [ ] The template contributor participated in a template working group to create this template.
+- [ ] This template was workshopped with the community.
+- [ ] A template mentor officially approved this template set to move to the pull request phase.
```
"Officially" isn't doing much work here. I'd remove it.

barbaricyawps replied:
Sure.

barbaricyawps replied:
I've adopted this suggestion.

aidanatwork commented on template-quality-checklist.md:
```diff
@@ -0,0 +1,102 @@
+- [ ] A template mentor officially approved this template set to move to the pull request phase.
+
+
+### Template set requirements
```
See my previous comments about the naming of these files.

barbaricyawps replied:
I've adopted the new names we agreed upon. Resolving.

aidanatwork commented on template-quality-checklist.md:
```diff
@@ -0,0 +1,102 @@
+
+- [ ] Check rendered Markdown output to ensure it renders correctly and follows the template markdown style guide.
+- [ ] The template title is present and is an H1.
+- [ ] The template includes an introductory comment pointing them to other template resources.
```
Is "pointing to other template resources" a requirement for the introductory comment? I did not think so, from the description provided earlier.

barbaricyawps replied:
My thinking here is that if they only land on the template file itself and don't realize there are supporting documents (like a guide), they need to know those supporting docs exist. I could make sure that's included in the template instructions.

barbaricyawps replied:
I've added some clarity to this section per your feedback. Resolving.

aidanatwork commented on template-quality-checklist.md:
```diff
@@ -0,0 +1,102 @@
+### Template example requirements (optional) - PULL REQUEST REVIEWER ONLY
+
+- [ ] Check rendered Markdown output to ensure it renders correctly and follows the template markdown style guide.
+- [ ] The template example provides is well-written and provides a clear example of how to use the template.
```
Typo here. Change "provides" to "provided".

barbaricyawps replied:
Sure.

barbaricyawps replied:
I've adopted this suggestion.

aidanatwork commented on template-roles-and-resources.md:
```diff
@@ -0,0 +1,167 @@
+The working groups that are specifically devoted to template writing are:
+
+- [Community Docs](https://thegooddocsproject.dev/working-group/community-docs/)
+- More forthcoming!
```
Should we add the APAC/EMEA group here?

barbaricyawps replied:
Yep, that group formed after I submitted this RFC.

barbaricyawps replied:
I adopted this suggestion.

aidanatwork commented on template-roles-and-resources.md:
```diff
@@ -0,0 +1,167 @@
+- Help connect writers to resources or subject matter experts that can provide feedback on their template drafts.
+- Provide assistance to contributors working on templates in their group, especially when they are blocked.
+- Coordinate and communicate regularly with the templates coordinator.
+- Work with the templates coordinator to ensure each template receives at least three community reviews on their draft when a template is in the workshop phase.
```
I would avoid the term "workshop phase" here in favor of "review phase".

barbaricyawps replied:
I think I've changed all instances of workshop phase to feedback phase. Resolving.

aidanatwork commented on template-roles-and-resources.md:
```diff
@@ -0,0 +1,167 @@
+
+- Aidan Doherty
+- Alyssa Rock
+- Cameron Shorter
```
Should Aaron be added to this list? Since he is templates coordinator?

barbaricyawps replied:
I don't think Aaron has plans to lead a template focused working group.

aidanatwork commented on template-roles-and-resources.md:
```diff
@@ -0,0 +1,167 @@
+To become a pull request reviewer:
+- You should be a member in good standing in the Good Docs Project community, as defined in our [Code of Conduct](https://thegooddocsproject.dev/code-of-conduct/).
+- You should be familiar with the Conventional Comments theory of labeling feedback.
+- You should be an experienced technical writer who is familiar with our project’s standards for quality.
```
thought: Experienced technical writer is an interesting term to consider. I would consider this no barrier to my participation. But compared to someone like Cameron, I might not be considered very experienced. I hope "impostor syndrome" or just not thinking of oneself as experienced isn't too much of a barrier for people to step up.

barbaricyawps replied:
You're going to be just fine, Aidan. :)

aidanatwork commented on template-roles-and-resources.md:
```diff
@@ -0,0 +1,167 @@
+- You should have experience with the template writing process, which is demonstrated by one or more of the following:
+  - Completing a full template set from beginning to end.
+  - Participating in a template writing working group for at least 2 months.
+- You must be nominated for this role by the existing templates coordinator or a member of the project steering committee.
```
question: Should template mentors be able to nominate pull request reviewers?

barbaricyawps replied:
Yes, I think that's reasonable.

barbaricyawps replied:
I've adopted this suggestion.

viraji reviewed (commented):
+1

johns10 commented on template-contributing-guide.md:
```diff
@@ -0,0 +1,341 @@
+  <tr>
+    <td>Template example</td>
+    <td>The Good Docs Project is in the process of designing a fake documentation project that can provide examples of our templates in action. Until this project is ready, it is optional to provide an example of the template. However, if you want to create a fake example as part of your template set, you may do so.</td>
+    <td>No, optional</td>
```
I personally think this should be required. It's literally my best resource when I'm looking at templates, because of the way my brain works. It should be a trivial ask.

barbaricyawps replied:
Unfortunately, it has to be optional since our example project is not available yet.

barbaricyawps replied:
I've noted this comment in the feedback section of the RFC for now.

camerons commented on rfc-005-template-contributing-process.md:
```diff
@@ -0,0 +1,154 @@
+
+- [ ] Add these files to the `templates` repository and ensure they are linked from the website
+- [ ] Nominate a member to fill the templates coordinator role
+- [ ] Nominate members to fill the template mentor roles
```
Suggestion (non-blocking): I'm feeling a bit nervous about pre-filling roles and titles.
I think that within a healthy volunteer community, people who are going to do the work (and do it well) typically don't step up until there is work needing to be done and they have the bandwidth and motivation at that point in time.
Yes, we definitely need a template coordinator now - this is an active role needing filling.
But I suggest we don't assign people to roles associated with development of future templates, and don't expect that just because someone mentored a template that they were interested in, that they will also mentor other templates too.
Instead, I think we should acknowledge that people will move in and out of these roles on an as needed basis.
Maybe on our template development landing page, we should include the roles of people involved in the template (which will also serve as a nice record and acknowledgement for their work).

barbaricyawps replied:
That's fair, Cameron. But I really can't see how we would operate without having clearly designated templates coordinator or template mentors in this process.

The same goes for pull request reviewers because those people will be given specific GitHub permissions that we won't freely distribute to everyone.

I can see the case for not defining community reviewers though. It can simply be anyone from the community.

barbaricyawps replied:
I'm going to note this concern in the feedback section of the RFC.

camerons commented on template-deliverables.md:
```diff
@@ -0,0 +1,145 @@
+  <tr>
+    <td>Raw template</td>
+    <td>The raw template is the basic template for the type of document you are creating. It provides a rough outline of the suggested content and a few embedded writing tips for how to fill in the different sections of the template.</td>
+    <td>Yes, required</td>
```
Nitpick (non-binding): I suggest being more concise. As this column heading says "Required?" I suggest:
1. Replace "Yes, required" with "required".
2. Replace "No, optional" with "optional".

barbaricyawps replied:
I've adopted this suggestion. Resolving.

camerons commented on rfc-005-template-contributing-process.md:
```diff
@@ -0,0 +1,154 @@
+6. **Improve the template with user feedback** - As users try the template in the wild, they may report usability issues or provide feedback for improvements to the template. Either the original template author or another templateer will evaluate feedback and incorporate it into future versions of the template. When that happens, the template will go through the previous phases again.
+
+Each template set will include:
+- **Raw template** - The basic template for the type of document you are creating. It provides a rough outline of the suggested content and a few embedded writing tips for how to fill in the different sections of the template.
```
Nitpick (non-blocking): The explanation text from here is a copy of "Template Deliverables". This is a situation where you are creating multiple-points-of-truth for the same concept, which is a dangerous thing when scaled out to a large management of requirements level, as the points-of-truth will inevitably get out of sync. Better is to just list the title, such as "raw template" and hyperlink to the point of truth.
Don't worry about fixing here. As this is a short term proposal, it is not going to be a problem. Just something to consider for future.

barbaricyawps replied:
Thanks, Cameron. I'll note this in the feedback section of the RFC.

camerons commented on template-deliverables.md:
```diff
@@ -0,0 +1,145 @@
+
+The raw template is the basic template for the type of document you are creating.
+It provides a rough outline of the suggested content and a few embedded writing tips for how to fill in the different sections of the template.
+The template user will copy this template and begin filling it in or editing it as their starting point in the writing process.
```
Suggestion (non binding): I think we should introduce a point or section that explains the key audience types that these documents are addressing:
1. The document author, who is the user of the template (as described here).
2. The document reader, who is the audience of the document author.
If we skip explaining these audiences, I feel our readers are likely to get confused. (Actually, they are probably going to get confused no matter what, but we can start to help them.)

barbaricyawps replied:
I agree with you, but I wonder if this should be a separate project-wide document (e.g. our personas). I think the Content Strategy working group should create this content and that they should be responsible for putting this on our website. Then we can link to it here.

For now I'll note your suggestion in the RFC feedback section.

camerons commented on template-deliverables.md:
```diff
@@ -0,0 +1,145 @@
+
+Each raw template should have this content:
+- **The template title** - The type of document this template is for. For example, "Tutorial template" or "README template."" This title should be formatted with the heading 1 weight.
+- **Introductory comment** - Include an embedded writing tip at the start that tells users they should first read the accompanying template guide and before-you-start guide before they fill in the template. Put introductory comments in {curly brackets}.
```
Suggestion: I've found myself using curly brackets in two ways in the [base template](https://github.com/thegooddocsproject/templates/blob/base-restructure3/base/base-template.md), and I think we should call them out separately:
1. As explanatory writing tips to the doc author. I suggest these be written with the keyword "Tip" (or similar):
{Tip: Some writing tip here.}
2. As text or variables to be replaced within a sentence, such as:
This document {explains/covers/shows what/how to do something}. It is designed to help {persona or personas} to achieve {some goal}.

camerons replied:
> :triangular_flag_on_post: I see that Alyssa has discovered how to use emojis in github text, along with a tab insert. This is pretty cool. (See this sentence for example).

I suggest that we reconsider how we do our call out tips, maybe use a combination of curly brackets and emojis.

barbaricyawps replied:
I think we need to create a template style guide and put stuff like this in that guide. I've had that on my to-do list for awhile. For now, I've noted your comments in the feedback for this RFC.

camerons commented on template-deliverables.md:
```diff
@@ -0,0 +1,145 @@
+
+### Content and formatting guidelines
+
+Each raw template should have this content:
```
Suggestion: I suggest breaking these bullet points into two lists:
1. Structure (using the titles you have)
2. Style (talking about how to use curly brackets, how deep to make headings, ...}

barbaricyawps replied:
I'm not sure how to take action on this comment. Can we defer it for after the RFC, perhaps as a pull request against this document in the templates repo? I'll note your feedback in the RFC feedback.

camerons commented on template-deliverables.md:
```diff
@@ -0,0 +1,145 @@
+- **The template sections** - Each section of the temple should begin with a heading with the heading 2 weight. The recommended content and some embedded writing tips fall under the headings.
+- **Embedded writing tips** - You can provide some lightweight writing tips that provide some tips or hints about what kind of content the template user might choose to put in a section of the document. Put embedded writing tips in {curly brackets}.
+
+> :triangular_flag_on_post: **NOTE: The basic content of the raw template depends on the type of document you are creating a template for. Different types of documents can have widely varying content needs. After you have completed your research into this document type and are preparing to create the template itself, you should create headings for each of the main sections of this type of documen ...
```
Suggestion: This statement has good foundations. Maybe tweak to:
> The content of the raw template depends on the type of document you are creating a template for. Different types of documents can have widely varying content needs. While similar document types are likely to require similar structures, other document types may be structured quite differently.

barbaricyawps replied:
I've adopted this suggestion.

camerons commented on template-deliverables.md:
```diff
@@ -0,0 +1,145 @@
+
+**Q: Is it better to include all the possible sections that someone could possibly include in the raw template or should it just have the most common sections?**
+
+A: While this is mostly a judgment call on your part as the template author, it might be better to err on the side of being comprehensive.
```
Suggestion (non-binding): This answer is acceptable for this first round, and I like the suggestion of making it a judgement call. I'd suggest that we actually articulate a preferred project size to help orient the template author. I've had a crack at this in [Using the base template set](https://docs.google.com/document/d/16abC0MNnq-dh77jCcQxwsFk8X7n-xwBVz7FTwnP_oO4/edit#heading=h.qcgdxta57ppz). Quote:
**Target project**
For the raw template and guide imagine you are writing for:
* An emerging project, getting ready to publish their first 1.0 stable release;
* With 1 to 5 developers;
* Which doesn’t have a dedicated technical writer;
* Which is considering an audience of tens to hundreds of users.

barbaricyawps replied:
I feel like this is beyond my pay grade to decide and that it needs further community discussion. I'll note it in the feedback section.

camerons commented on template-deliverables.md:
```diff
@@ -0,0 +1,145 @@
+
+In essence, this guide supports the key decisions that the template user might need to make during the writing process.
+
+This guide might also optionally include a checklist of the required components for this type of document.
```
Suggestion: At the moment, we have a separate checklist file. I'd suggest we either include the checklist in this guide, or we have it as an external doc, but not both. My preference is to have the checklist separate, and I believe the checklist can stand alone as its own document. (A reviewer can copy the checklist and tick off the boxes separately to needing to read the guide.)
Side note: An example of a checklist for another project I've worked on: https://wiki.osgeo.org/wiki/OSGeoLive_AddProject#Quickstart_Review_Checklist

barbaricyawps replied:
I'm not sure that I understand this comment well enough to take action on it. I'll note it in the feedback section of the RFC.

camerons commented on template-deliverables.md:
```diff
@@ -0,0 +1,145 @@
+
+## Template guide
+
+Example filename: abc-template-guide.md
```
Observation: We previously called this file abc-guide.md . Changing to abc-template-guide.md is a bit more of a mouthful, but I can see how it could add clarity, so like the idea.

barbaricyawps replied:
Great!

camerons commented on template-deliverables.md:
```diff
@@ -0,0 +1,145 @@
+
+- **Template guide title** - The type of document this template is for and the words "guide." For example, "Tutorial guide" or "README guide." This title should be formatted with the heading 1 weight.
+- **Introductory comment** - Include an embedded writing tip at the start that tells users they should first read this guide and the before-you-start guide before they fill in the template. This comment could also mention how to get the raw template file. Put introductory comments in {curly brackets}.
+- **Why do I need this type of document?** - Provide some of the information about the purpose of this type of document and how it helps your documentation project. The header for this section should be formatted with the heading 2 weight.
```
Suggestion (considered important):
I feel we should have a placeholder in here to put some of the content (which is currently in the [base theory document](https://github.com/thegooddocsproject/templates/blob/dev/base/base-theory.md)).
I'd suggest adding:
* H2: About this document type
* H3: Why do I need this type of document?
* H3: ... {Place holder for other sections, which we can add as we become more mature}

For the types of headings to include, consider titles from the [base-theory](https://github.com/thegooddocsproject/templates/blob/dev/base/base-theory.md) doc
* User stories
* Nature of content
* Priorities for each quality criteria
* Implementation strategy (e.g. can this doctype be automatically generated from code?)
* Business case
* When to include? When not to include?
* Customization guidelines
* Maintenance strategy
* Backing research and further reading

barbaricyawps replied:
Sounds good. I'll work it in.

barbaricyawps replied:
I've adopted this suggestion.

camerons commented on template-deliverables.md:
```diff
@@ -0,0 +1,145 @@
+- **Introductory comment** - Include an embedded writing tip at the start that tells users they should first read this guide and the before-you-start guide before they fill in the template. This comment could also mention how to get the raw template file. Put introductory comments in {curly brackets}.
+- **Why do I need this type of document?** - Provide some of the information about the purpose of this type of document and how it helps your documentation project. The header for this section should be formatted with the heading 2 weight.
+- **Content of this template** - This header marks the beginning of the part of the guide that explains how to fill in each section of the guide. It should be formatted with the heading 2 weight.
+- **About the {insert section name here} section** - Next, include each section as they appear in the template with the heading 3 weight. For example, if your template includes a "Before you start" section, this heading should say "About the before your start section." And then it should provide guidance about what kind of content goes in that section and why that section might be included in the ...
```
* I think that some headings won't need extra explanation. From the following options, I'd err toward 1.
1. Don't add an About {section} for this section
2. Include the About {section} so the reader knows that the section hasn't been forgotten, but say "No extra guidance".
* Some sections can have a light weight version or a heavy version, depending on the size of the project. This can be discussed here.

barbaricyawps replied:
For reference, I'm basing this off of my style guide's about document: https://github.com/thegooddocsproject/templates/blob/dev/style-guide/about-style-guide.md

That was the format I used there and it communicates well, in my opinion.

camerons commented on template-deliverables.md:
```diff
@@ -0,0 +1,145 @@
+- **Why do I need this type of document?** - Provide some of the information about the purpose of this type of document and how it helps your documentation project. The header for this section should be formatted with the heading 2 weight.
+- **Content of this template** - This header marks the beginning of the part of the guide that explains how to fill in each section of the guide. It should be formatted with the heading 2 weight.
+- **About the {insert section name here} section** - Next, include each section as they appear in the template with the heading 3 weight. For example, if your template includes a "Before you start" section, this heading should say "About the before your start section." And then it should provide guidance about what kind of content goes in that section and why that section might be included in the ...
+- **Checklist (optional)** - If it makes sense to include a checklist of the typical content required for this type of document, include that here.
```
Suggestion: I think it is a bit much to include checklists at the section level. I'd be inclined to have all checklists managed in the external checklist template.
Maybe indicate that this is a H3 level.

barbaricyawps replied:
The Community Docs group didn't like the idea of having an external checklist for each template. See the notes for August 5: https://docs.google.com/document/d/1iRxB0T7FoVjIeG_LawbCzcLfXCILlpFdzn-4BPDGR9Y/edit?usp=sharing

camerons commented on template-deliverables.md:
```diff
@@ -0,0 +1,145 @@
+- **Content of this template** - This header marks the beginning of the part of the guide that explains how to fill in each section of the guide. It should be formatted with the heading 2 weight.
+- **About the {insert section name here} section** - Next, include each section as they appear in the template with the heading 3 weight. For example, if your template includes a "Before you start" section, this heading should say "About the before your start section." And then it should provide guidance about what kind of content goes in that section and why that section might be included in the ...
+- **Checklist (optional)** - If it makes sense to include a checklist of the typical content required for this type of document, include that here.
+- **Additional resources (optional)** - If you’d like to point the user to any helpful guides for deeper study or if you want to acknowledge any resources that inspired you while you wrote this template, link to or cite those resources here.
```
Maybe suggest this is a H3 heading.

barbaricyawps replied:
What would its parent heading be?

camerons reviewed (commented):
Added review for template deliverables. Overall feeling is that this is really good material. Thanks for bringing this far.

barbaricyawps commented:
Thank you kindly to everyone who commented on this RFC up to this point with your excellent suggestions and feedback. I've made revisions to the documents based on all the feedback I've received here. In most cases, the feedback has been adopted and added into the RFC and supporting documents. In rare cases where the feedback wasn't incorporated, I've added it to the RFC feedback section for further action and discussion as needed.

I believe this RFC is ready to move into the final comment and voting period now.

camerons commented on template-contributing-guide.md:
```diff
@@ -0,0 +1,344 @@
+  <tr>
+    <td>4 - Get feedback on drafts from the community</td>
+    <td><ul>
+          <li>Revise and refine the drafts of your deliverables based on feedback from at least three community reviewers. Community reviewers are anyone who is a member of the Good Docs Project, including members of template writing working groups.</li>
```
Suggestion: non-blocking: My recollection of our last meeting was that we have a "strong preference for three or more reviewers" and not make it a hard requirement.
```suggestion
          <li>Revise and refine the drafts of your deliverables. Expect to receive feedback from at least three community reviewers. Community reviewers are anyone who is a member of the Good Docs Project, including members of template writing working groups.</li>

```

barbaricyawps commented:
+1 from me in case that needs to be spelled out.

camerons commented:
+1 from me for this proposal. This has been a huge effort from you Alyssa to bring this forward. Thank you so much for such excellent work.

aidanatwork commented:
+1 for this latest version of the RFC

carriecrowe1138 commented:
+1! Alyssa, thank you for your great work!

bwklein commented:
+1 - Bryan

emckean commented:
+1 from me!

mgan59 commented:
__+1__ from me

techtrailhead commented:
+1 from me!

viraji commented:
+1 from me too! Not sure if there was a mail, but I reviewed this with my
comments last week.

Thanks,
/Viraji.



On Wed, Sep 29, 2021 at 9:28 AM Ankita Tripathi ***@***.***>
wrote:

> +1 from me!
>
> —
> You are receiving this because you commented.
> Reply to this email directly, view it on GitHub
> <https://github.com/thegooddocsproject/request-for-comment/pull/10#issuecomment-929811023>,
> or unsubscribe
> <https://github.com/notifications/unsubscribe-auth/AGK37RS74OVJ7FPSJKWOUHTUEKFFJANCNFSM5CGWXCAA>
> .
> Triage notifications on the go with GitHub Mobile for iOS
> <https://apps.apple.com/app/apple-store/id1477376905?ct=notification-email&mt=8&pt=524675>
> or Android
> <https://play.google.com/store/apps/details?id=com.github.android&referrer=utm_campaign%3Dnotification-email%26utm_medium%3Demail%26utm_source%3Dgithub>.
>
>

acpkendo commented:
+1 from me, great work on this Alyssa!
